### DISCUSSION TITLE
IAT plugin problems!

### DISCUSSION THREAD

**Discussion by margin-bqk** [jspsych/jsPsych#1898] [Q&A] [answered]
Hello!
I noticed that when I set display_feedback in the IAT plugin to false, the subjects will still have to **press a key**  after pressing the wrong key to move on to the next trail.
I was planning to not giving the subjects the feedback and just don't let them know their result in every trail.
        

var con_train_block = {
            type: 'iat-html',
            stimulus:jsPsych.timelineVariable('sn'),
            stim_key_association:jsPsych.timelineVariable('sn_ans'),
            bottom_instructions: '<p>If you press the wrong key, a red X will appear. Press the other key to continue</p>',
            trial_duration: 3000, //Only if display_feedback is false
            display_feedback:false,
            left_category_key: 'f',
            right_category_key: 'j',
            left_category_label: ['数学和男性名字'],
            right_category_label: ['语文和女性名字'],   
            response_ends_trial: true,
        }

The doc says that when the display_feedback is set to false, trail_duration is needed and the trail will continue automatically, but it still need key press!

Needing help to help me finishing the final assignment lol.
Huge thanks!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @margin-bqk, yes I just tried this and I see what you mean. I think the trial should end immediately after an incorrect response, with the following parameters:
```js
var trial = {
  type: 'iat-image',
  force_correct_key_press: false,
  display_feedback: false,
  response_ends_trial: true,
  // other parameters...
}
```
With these settings, I'm finding that a correct response ends the trial immediately, but an incorrect response doesn't. Instead, I think the participant is required to press another key to end the trial. I've created an Issue for this (#1900) so that we can verify that this is a bug, and if so, fix it. 

In the meantime, you can fix this yourself with the following modifications to the IAT plugin:
https://github.com/jspsych/jsPsych/blob/ae332b33477d686388bafbd03b0c8e1c9945999c/plugins/jspsych-iat-html.js#L219-L223
Replace the keyboard listener in the conditional (lines 220-222) above with:
```js
end_trial();
```

https://github.com/jspsych/jsPsych/blob/ae332b33477d686388bafbd03b0c8e1c9945999c/plugins/jspsych-iat-html.js#L251-L255
Replace the keyboard listener in the conditional (lines 252-254) above with:
```js
end_trial();
```

**Reply by margin-bqk** [reply]
Huge thanks to you. Now it perfectly works. And I will be able to finish my assignment. Wish jsPsych be more perfect!
